keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -195,26 +202,13 @@ def call_with_cache(
             the final hidden representation of the input tokens, and `cache` is
             the decoding cache.
         """
-        tokens = self.backbone.token_embedding(token_ids)
-        positions = self.backbone.position_embedding(
-            tokens, start_index=cache_update_index
+        return self.backbone_with_cache(
+            {
+                "token_ids": token_ids,
+                "cache": cache,
+                "cache_update_index": ops.convert_to_tensor(cache_update_index),
```

Comment:
Why do we need `conver_to_tensor` here?

Reply:
Not sure why. It would not work without it.